tests/unit/load-webpack.ts

```diff
@@ -0,0 +1,145 @@
+import * as assert from 'intern/chai!assert';
+import * as registerSuite from 'intern!object';
+import global from '../../src/global';
+import load from '../../src/load-webpack';
+
+interface WebpackModules {
+	[id: number]: any;
+}
+
+let webpackModules: WebpackModules;
+function setModules(modules?: { [mid: string]: any }) {
+	let nextId = 0;
+	webpackModules = Object.create(null) as WebpackModules;
+
+	if (modules) {
+		const idMap = Object.create(null);
+		Object.keys(modules).forEach((mid: string) => {
+			const id = nextId++;
+			idMap[mid] = { id, lazy: mid.indexOf('bundle!') > -1 };
```

Comment:
One option would be to inject a `__loaders__` object that gets passed the `require` function, `mid`, and `resolve` callback, with `load` reducing the value for multiple loaders as needed. We could provide `bundle` out of the box, but others would need to be specified via a config. Are there any security considerations that would need to be addressed with such an implementation?